sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryRequestFailedDetailsParser.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Containers.ContainerRegistry
+{
+    internal class ContainerRegistryRequestFailedDetailsParser : RequestFailedDetailsParser
+    {
+        public override bool TryParse(Response response, out ResponseError error, out IDictionary<string, string> data)
+        {
+            error = null;
+            data = null;
+
+            if (response.ContentStream == null)
+            {
+                return false;
+            }
+
+            BinaryData content = BinaryData.FromStream(response.ContentStream);
+            if (content.ToMemory().Length == 0)
+            {
+                return false;
+            }
+
+            // Buffer the stream to log the content in the message
+            // See: https://github.com/Azure/azure-sdk-for-net/issues/35355
+            MemoryStream stream = new(content.ToArray());
```

Comment:
I think we can simply extract this part of the `ResponseBodyPolicy` into helper method: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs#L78-L110

Something like
```cs
internal static async ValueTask BufferResponse(Response response, bool async, CancellationTokenSource cts)
{
    var responseContentStream = message.Response.ContentStream;
    if (responseContentStream is null or MemoryStream || responseContentStream is MemoryStream)
    {
        return;
    }
    try
    {
        var bufferedStream = new MemoryStream();
        if (async)
        {
            await CopyToAsync(responseContentStream, bufferedStream, cts).ConfigureAwait(false);
        }
        else
        {
            CopyTo(responseContentStream, bufferedStream, cts);
        }

        responseContentStream.Dispose();
        bufferedStream.Position = 0;
        message.Response.ContentStream = bufferedStream;
    }
    // We dispose stream on timeout or user cancellation so catch and check if cancellation token was cancelled
    catch (Exception ex)
        when (ex is ObjectDisposedException
                  or IOException
                  or OperationCanceledException
                  or NotSupportedException)
    {
        ThrowIfCancellationRequestedOrTimeout(oldToken, cts.Token, ex, networkTimeout);
        throw;
    }
}
```

Reply:
And then call it here with `BufferResponse(response, false, new CancellationTokenSource()).EnsureCompleted()`;